### DISCUSSION TITLE
router.add_api_route does not take the response model from the function return type hint by default(the defult is None), is this intended?

### DISCUSSION THREAD

**Discussion by jeffective** [fastapi/fastapi#10726] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def test() -> dict:
    return {'test_key':'test_value'}


router = APIRouter()
self.router.add_api_route('/test', test, methods=['GET'], response_model=dict) #requires setting response_model here, does not get it from the return type hint
```

### Description

I would expect add_api_route to grab the return type hint for response model by default

### Operating System

Linux

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

3.11.6

**Comment by jeffective** [comment]
its beginning to look like add_api_route doesn't add the response model at all... I can't get this schema to show up!

**Comment by jeffective** [comment]
I was only able to get the schema to show up by using the responses parameter, otherwise it doesn't show the schemas at all.

```python

class SingleCommanderLockStateResponseModel(BaseModel):
    is_locked: bool
    username: Optional[str] = None #username of commander
    message: str = "" #message from commander to other users
    locked_since: Optional[datetime] = None #time since lock last refreshed
    timeout: timedelta = timedelta(seconds=0) #time that lock expires after locked_since

#doesnt work:
self.router.add_api_route('/check_my_auth', self.check_my_auth, methods=['GET'], response_model=SingleCommanderLockStateResponseModel)

#works:
self.router.add_api_route('/check_my_auth', self.check_my_auth, methods=['GET'], responses={200: {'model': SingleCommanderLockStateResponseModel}})
```

**Comment by YuriiMotov** [comment]
Can't reproduce this..
Works fine with both (FastAPI 0.104, Pydantic 2.5.2) and current versions (FastAPI 0.115.12, Pydantic 2.11.5).
Probably related to one of sub-dependencies.

```python
from fastapi import APIRouter, FastAPI
from pydantic import BaseModel

app = FastAPI()

class SingleCommanderLockStateResponseModel(BaseModel):
    is_locked: bool

router = APIRouter()

# ==================================================================================
async def check_my_auth() -> SingleCommanderLockStateResponseModel:
    return

router.add_api_route("/check_my_auth1", check_my_auth, methods=["GET"])


# ==================================================================================
async def check_my_auth2():
    return

router.add_api_route(
    "/check_my_auth2",
    check_my_auth2,
    methods=["GET"],
    response_model=SingleCommanderLockStateResponseModel,
)

# ==================================================================================
async def check_my_auth3():
    return

router.add_api_route(
    "/check_my_auth3",
    check_my_auth3,
    methods=["GET"],
    responses={200: {"model": SingleCommanderLockStateResponseModel}},
)


# ==================================================================================
app.include_router(router=router)

```
